config/src/converters/k8s/gateway_config.rs

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+use k8s_intf::gateway_agent_crd::GatewayAgent;
+
+use crate::DeviceConfig;
+use crate::converters::k8s::FromK8sConversionError;
+use crate::external::overlay::Overlay;
+use crate::external::underlay::Underlay;
+use crate::external::{ExternalConfig, ExternalConfigBuilder};
+
+/// Convert from `GatewayAgent` (k8s CRD) to `ExternalConfig` with default values
+impl TryFrom<&GatewayAgent> for ExternalConfig {
+    type Error = FromK8sConversionError;
+
+    fn try_from(ga: &GatewayAgent) -> Result<Self, Self::Error> {
+        let name = ga
+            .metadata
+            .name
+            .as_ref()
+            .ok_or(FromK8sConversionError::MissingData(
+                "metadata.name not found".to_string(),
+            ))?
+            .as_str();
+        let device = DeviceConfig::try_from(ga)?;
+        let underlay = Underlay::try_from(ga.spec.gateway.as_ref().ok_or(
+            FromK8sConversionError::MissingData(format!(
+                "gateway section not found in spec for {name}"
+            )),
+        )?)?;
+        let overlay = Overlay::try_from(&ga.spec)?;
+        let Some(gen_id) = ga.metadata.generation else {
```

Comment:
This check could have been made much before, without needing to call any try_from.